src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -689,24 +694,26 @@ public virtual HtmlString GenerateAntiforgery([NotNull] ViewContext viewContext)
 
                     if (!string.IsNullOrEmpty(errorText))
                     {
-                        var listItem = new TagBuilder("li", _htmlEncoder);
+                        var listItem = new TagBuilder("li");
                         listItem.SetInnerText(errorText);
-                        htmlSummary.AppendLine(listItem.ToString(TagRenderMode.Normal));
+                        htmlSummary.Append(listItem.ToHtmlContent(TagRenderMode.Normal));
+                        htmlSummary.Append(Environment.NewLine);
                     }
                 }
             }
 
-            if (htmlSummary.Length == 0)
+            if (htmlSummary.ToString().Length == 0)
```

Comment:
Suggest we need something like `TagHelperOutput`s `IsNullOrEmpty` in either `IHtmlContent` or `BufferedHtmlContent`. This less about performance (though it's not great to create a `string` only to immediately throw it away) than straightforwardness.